tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -0,0 +1,518 @@
+		if err != nil {
+			log.Info("failed to wait for instance: %s to terminate", instanceID)
+		}
+	}
+	// Delete security groups after associated instances are terminated.
+	for _, sgID := range instances.SecurityGroupIDs {
+		sgInUse, err := a.isSGInUse(sgID)
+		if err != nil {
+			log.Info("failed to check if security group %s is in use, %v", sgID, err)
+		}
+		if !sgInUse {
+			err = a.deleteSG(sgID)
+			if err != nil {
+				log.Info("failed to delete Security Group: %s", sgID)
+			}
+		}
+	}
+	err = resource.RemoveInstallerInfo(instances, a.resourceTrackerDir)
+	if err != nil {
+		log.Info("%s file is not updated, %v", resource.InstallerInfoFileName, err)
+	}
+	return nil
+}
+
+// createInstances creates one VM instance based on the given information and returns if no instance is created.
+func (a *awsProvider) createInstances(imageID, instanceType, sshKey, subnetID, sgID, workerSgID string,
+	iamProfile *ec2.IamInstanceProfileSpecification) (string, error) {
+	runResult, err := a.EC2.RunInstances(&ec2.RunInstancesInput{
+		ImageId:            aws.String(imageID),
+		InstanceType:       aws.String(instanceType),
+		KeyName:            aws.String(sshKey),
+		SubnetId:           aws.String(subnetID),
+		MinCount:           aws.Int64(1),
+		MaxCount:           aws.Int64(1),
+		IamInstanceProfile: iamProfile,
+		SecurityGroupIds:   aws.StringSlice([]string{sgID, workerSgID}),
+	})
+	if err != nil {
+		return "", err
+	}
+	if len(runResult.Instances) < 1 {
+		return "", fmt.Errorf("failed to create an instance")
+	}
+	return *runResult.Instances[0].InstanceId, nil
+}
+
+// findOrCreateSg tries to find a Security Group based on VPC ID and Group Name. If no such Security Group exists,
+// it creates a Security Group under the VPC with a Group Name '<infraID>-sg'.
+// The function returns Security Group ID or error for both finding and creating a Security Group.
+func (a *awsProvider) findOrCreateSg(infraID, vpcID string) (string, error) {
+	sgName := strings.Join([]string{infraID, "sg"}, "-")
+	sgs, err := a.EC2.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("vpc-id"),
+				Values: aws.StringSlice([]string{vpcID}),
+			},
+			{
+				Name:   aws.String("group-name"),
+				Values: aws.StringSlice([]string{sgName}),
+			},
+		},
+	})
+	if err != nil {
+		return "", fmt.Errorf("error finding Security Group %v", err)
+	} else if len(sgs.SecurityGroups) > 0 {
+		return *sgs.SecurityGroups[0].GroupId, nil
+	}
+	sg, err := a.EC2.CreateSecurityGroup(&ec2.CreateSecurityGroupInput{
+		GroupName:   aws.String(sgName),
+		Description: aws.String("security group for RDP and all traffic within VPC"),
+		VpcId:       aws.String(vpcID),
+	})
+	if err != nil {
+		return "", fmt.Errorf("failed to create Security Group %v", err)
+
+	}
+	return *sg.GroupId, nil
+}
+
+// authorizeSgIngress attaches inbound rules for RDP from user's external IP address and all traffic within the VPC.
```

Comment:
You cannot modify the existing worker SG. You need to create a new one and apply these permissions to it. In addition you need to find the existing worker SG and attach it to the VM.